astropy/units/core.py

```diff
@@ -2684,6 +2684,9 @@ def _condition_arg(value):
     if isinstance(value, (np.ndarray, float, int, complex, np.void)):
         return value
 
+    if hasattr(value, "__array_ufunc__"):
```

Comment:
This may work but seems a bit haphazard. I guess basically we want to test whether something is "array like". I think the correct way to do this would be to use check for `__array_namespace__` -- https://data-apis.org/array-api/latest/purpose_and_scope.html#how-to-adopt-this-api (but possibly @nstarman can advice).

Indeed, eventually we would be able to replace the `isinstance(value, np.ndarray)` with that! (but not yet, older numpy do not have it).

Reply:
Agreed. `__array_namespace__` is the recommended approach.